jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -151,10 +158,18 @@ public void internalTrigger(final NodeInstance from, String type) {
                             .activateRuleFlowGroup(getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId());
 
                 } else {
+                    int fireLimit = DEFAULT_FIRE_RULE_LIMIT;
+                    
+                    if (inputs.containsKey(FIRE_RULE_LIMIT_PARAMETER)) {
+                        fireLimit = Integer.parseInt(inputs.get(FIRE_RULE_LIMIT_PARAMETER).toString());
+                    }
                     ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
                             .activateRuleFlowGroup(getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId());
 
-                    ((KieSession) getProcessInstance().getKnowledgeRuntime()).fireAllRules();
+                    int fired = ((KieSession) getProcessInstance().getKnowledgeRuntime()).fireAllRules(fireLimit);
+                    if (fired == fireLimit) {
+                        throw new RuntimeException("Fire rule limit reached " + fireLimit);
```

Comment:
Any chance we could include the property in the exception message, or some docs pointer? If I saw this exception, I would immediately start wondering what this limit is and where to set it.

Reply:
+1 here to provide more meaningful message